File: app/src/main/java/kr/djspi/pipe01/sql/SuperviseDao.java
package kr.djspi.pipe01.sql;

import androidx.room.Dao;
import androidx.room.Insert;
import androidx.room.OnConflictStrategy;
import androidx.room.Query;

import java.util.List;

@Dao
public interface SuperviseDao {

    @Query("SELECT * FROM Supervise ORDER BY supervise ASC")
    List<Supervise> getAll();

    @Query("SELECT supervise FROM Supervise WHERE id IN (:userId) LIMIT 1")
    String selectById(int userId);

    @Query("SELECT id FROM Supervise WHERE supervise IN (:supervise) LIMIT 1")
    int selectBySupervise(String supervise);

    @Insert(onConflict = OnConflictStrategy.REPLACE)
    void insertAll(Supervise... entity);

    @Insert(onConflict = OnConflictStrategy.REPLACE)
    void insert(Supervise entity);
}

File: app/src/main/java/kr/djspi/pipe01/sql/SuperviseDao.kt
package kr.djspi.pipe01.sql

import androidx.room.Dao
import androidx.room.Insert
import androidx.room.OnConflictStrategy
import androidx.room.Query

@Dao
interface SuperviseDao {

    @Query("SELECT * FROM Supervise ORDER BY supervise ASC")
    fun getAll(): List<Supervise>

    @Query("SELECT supervise FROM Supervise WHERE id = :userId LIMIT 1")
    fun selectById(userId: Int): String

    @Query("SELECT id FROM Supervise WHERE supervise = :supervise LIMIT 1")
    fun selectBySupervise(supervise: String?): Int

    @Insert(onConflict = OnConflictStrategy.REPLACE)
    fun insertAll(vararg entity: Supervise)

    @Insert(onConflict = OnConflictStrategy.REPLACE)
    fun insert(entity: Supervise)
}
